app/Http/Controllers/AuthController.php

```diff
@@ -53,21 +62,96 @@ public function register(RegisterRequest $request): JsonResponse
         }
     }
 
+    /**
+     * Log in the user.
+     *
+     * @param LoginRequest $request The login request.
+     * @return JsonResponse The JSON response.
+     */
     public function login(LoginRequest $request): JsonResponse
     {
         $credentials = $request->validated();
-
-        if (Auth::attempt($credentials)) {
+        // Attempt to authenticate the user with the provided credentials
+        if (Auth::attempt($credentials, true)) {
+            // If authentication is successful, retrieve the authenticated user
             $user = Auth::user();
+            // Generate an access token using Laravel Passport
             $token = $user->createToken('authToken')->accessToken;
-
+            // Log successful login
             Log::info('User logged in successfully: ' . $request->email);
-
+            // Return a success response with the token and HTTP status code 200 (HTTP_OK)
             return response()->json(['token' => $token], Response::HTTP_OK);
         }
-
+        // If authentication fails, return an error response with HTTP status code 401 (HTTP_UNAUTHORIZED)
         return response()->json(['error' => 'Unauthorized'], Response::HTTP_UNAUTHORIZED);
     }
 
+    /**
+     * Reset user password.
+     *
+     * @param ResetPasswordRequest $request The reset password request.
+     * @return JsonResponse The JSON response.
+     */
+    public function resetPassword(ResetPasswordRequest $request): JsonResponse
+    {
+        $request->validate([
```

Comment:
remove the 'validate' code. Params are already validated in the custom request. our request first passes through a custom request where it is validated, only after that it gets to the controller